front-end/src/components/newsfeed-view/news-list/news-card/NewsCardSwitcher.js

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import NewsCard from './NewsCard';
+import InstagramNewsCard from './InstagramNewsCard';
+import TwitterNewsCard from './TwitterNewsCard';
+import FacebookNewsCard from './FacebookNewsCard';
+
+
+
+const NewsCardSwitcher = ({type, data}) => {
+    console.log(type);
+    switch (type) {
```

Comment:
as I suggested. A generic card able to handle different sources of data looks more attractive than this.